### PR TITLE
PHP 8.1: fix passing null to non-nullable deprecation

### PR DESCRIPTION
**Pull request by mustafauysal**
It seems `rtrim()` expects to be passed a non-nullable string on PHP 8.1. Passing `null` results `rtrim(): Passing null to parameter #1 ($string) of type string is deprecated` notice.

Ref:
* https://wiki.php.net/rfc/deprecate_null_to_scalar_internal_arg

### CODE REVIEW

**Comment by lomeraniel** [comment]
Noting here that a VIP customer is facing this issue: 

```
"rtrim(): Passing null to parameter #1 ($string) of type string is deprecated"

Trace from QM:
wp-includes/formatting.php:2827
rtrim()
wp-includes/formatting.php:2827
untrailingslashit()
wp-includes/formatting.php:2812
trailingslashit()
wp-content/mu-plugins/http-concat/concat-utils.php:27
WPCOM_Concat_Utils::realpath()
wp-content/mu-plugins/http-concat/cssconcat.php:84
WPcom_CSS_Concat->do_items()
wp-includes/functions.wp-styles.php:68
wp_print_styles()
wp-includes/class-wp-hook.php:310
do_action('wp_head')
wp-includes/general-template.php:3053
wp_head()
wp-content/themes/dispatch/header.php:48
load_template('wp-content/themes/dispatch/header.php')
wp-includes/template.php:720
locate_template()
wp-includes/general-template.php:48
get_header()
wp-content/themes/dispatch/front-page.php:15
```

**Comment by mustafauysal** [comment]
@lomeraniel could you give it a try on the root domain without having a trailingslash? ( e.g. `example.com` without `/` at the end )

**Comment by bytesizeguru** [comment]
Another VIP customer reporting the same issue today:

```
Deprecated rtrim(): Passing null to parameter [#1](https://wordpressvip.zendesk.com/agent/tickets/1) ($string) of type string is deprecated 183
wp-includes/formatting.php:2827
MU Plugin: http-concat
wp-includes/formatting.php:2827
rtrim()
wp-includes/formatting.php:2827
untrailingslashit()
wp-includes/formatting.php:2812
trailingslashit()
wp-content/mu-plugins/http-concat/concat-utils.php:27
WPCOM_Concat_Utils::realpath()
wp-content/mu-plugins/http-concat/cssconcat.php:84
WPcom_CSS_Concat->do_items()
wp-includes/script-loader.php:2244
print_admin_styles()
wp-includes/class-wp-hook.php:310
do_action('admin_print_styles')
wp-admin/admin-header.php:132
```

**Comment by rodruiz** [comment]
VIP customer here. We are having the same issue on our dev environment.

```log
wp-includes/formatting.php:2827
rtrim()
wp-includes/formatting.php:2827
untrailingslashit()
wp-includes/formatting.php:2812
trailingslashit()
wp-content/mu-plugins/http-concat/concat-utils.php:27
WPCOM_Concat_Utils::realpath()
wp-content/mu-plugins/http-concat/cssconcat.php:84
WPcom_CSS_Concat->do_items()
wp-includes/functions.wp-styles.php:68
wp_print_styles()
wp-includes/class-wp-hook.php:310
do_action('wp_head')
wp-includes/general-template.php:3053
wp_head()
wp-content/themes/rjth2019/header.php:44
load_template('wp-content/themes/rjth2019/header.php')
wp-includes/template.php:720
locate_template()
wp-includes/general-template.php:48
get_header()
wp-content/themes/rjth2019/single-post.php:21
```

**Review by emreerkan — APPROVED** [review]
LGTM

File: concat-utils.php [hunk 1]
```diff
@@ -24,7 +24,7 @@ public static function realpath( $url, $site_url ) {
 		$url_path = parse_url( $url, PHP_URL_PATH );
 		$site_url_path = parse_url( $site_url, PHP_URL_PATH );
 		// To avoid partial matches; subdir install at `/wp` would match `/wp-includes`
-		$site_url_path = trailingslashit( $site_url_path );
+		$site_url_path = is_null( $site_url_path ) ? '/' : trailingslashit( $site_url_path );
```
Comment (rebeccahum) [inline concat-utils.php:27]: Can we just return an empty string instead of `/` since that's what was the return value in previous PHP versions?

**Review by rebeccahum — CHANGES_REQUESTED** [review]

File: concat-utils.php (hunk 1 above)
Comment (rebeccahum) [reply concat-utils.php:27]:
```suggestion
		$site_url_path = trailingslashit( $site_url_path ?? '' );
```

**Review by rebeccahum — APPROVED** [review]

**Comment by rebeccahum** [comment]
Reproduced, fix works thanks

**Pull request merged**